File: service/adapter/CICDAdapter.js
const DataSourceAdapter = require('./DataSourceAdapter');

class CICDAdapter extends DataSourceAdapter {
  constructor(options = {}) {
    super();
    this.env = options.env || process.env;
  }

  fetch() {
    // Return minimal CI/CD related environment variables
    const keys = ['CI', 'GITHUB_ACTIONS', 'TRAVIS'];
    const data = {};
    keys.forEach(k => {
      if (this.env[k]) data[k] = this.env[k];
    });
    return data;
  }
}

module.exports = CICDAdapter;

File: dependency-mapper/dependency-mapper/src/main/java/com/example/mapper/adapter/CICDAdapter.java
package com.example.mapper.adapter;

import java.util.HashMap;
import java.util.Map;

public class CICDAdapter implements DataSourceAdapter<Map<String, String>> {
    private final Map<String, String> env;

    public CICDAdapter(Map<String, String> env) {
        this.env = env;
    }

    @Override
    public Map<String, String> fetch() {
        String[] keys = {"CI", "GITHUB_ACTIONS", "TRAVIS"};
        Map<String, String> data = new HashMap<>();
        for (String k : keys) {
            if (env.containsKey(k)) {
                data.put(k, env.get(k));
            }
        }
        return data;
    }
}
